cpp-client/deephaven/dhclient/include/public/deephaven/client/utility/table_maker.h

```diff
@@ -27,448 +34,349 @@
 
 namespace deephaven::client::utility {
 namespace internal {
-class TypeConverter {
-public:
-  template<typename T>
-  [[nodiscard]]
-  static TypeConverter CreateNew(const std::vector<T> &values);
-
-  template<typename T, typename GetValue, typename IsNull>
-  [[nodiscard]]
-  static TypeConverter CreateNew(const GetValue &get_value, const IsNull &is_null,
-      size_t size);
-
-  TypeConverter(std::shared_ptr<arrow::DataType> data_type, std::string deephaven_type,
-      std::shared_ptr<arrow::Array> column);
-  ~TypeConverter();
-
-  [[nodiscard]]
-  const std::shared_ptr<arrow::DataType> &DataType() const { return dataType_; }
-  [[nodiscard]]
-  std::shared_ptr<arrow::DataType> &DataType() { return dataType_; }
-
-  [[nodiscard]]
-  const std::string &DeephavenType() const { return deephavenType_; }
-  [[nodiscard]]
-  std::string &DeephavenType() { return deephavenType_; }
+template<typename T>
+struct ColumnBuilder {
+  // The below assert fires when this class is instantiated; i.e. when none of the specializations
+  // match. It needs to be written this way (with "is_same<T,T>") because for technical reasons it
+  // needs to be dependent on T, even if degenerately so.
```

Comment:
The "generately so" part is hard to understand I think, could we reword that last part here?

Reply:
Yeah I hear you. I rewrote this more verbosely to explain the motivation behind it. PTAL